server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Quotas.scala

```diff
@@ -59,4 +68,123 @@ case class Quota(quota: Map[Quotas.Type, Value]) extends AnyVal
 
 case class Value(used: UnsignedInt, max: Option[UnsignedInt])
 
-case class Quotas(quotas: Map[QuotaId, Quota]) extends AnyVal
\ No newline at end of file
+case class Quotas(quotas: Map[QuotaId, Quota]) extends AnyVal
+
+case class UnparsedQuotaId(id: Id)
+
+case class QuotaIds(value: List[UnparsedQuotaId])
+
+case class QuotaGetRequest(accountId: AccountId,
+                           ids: Option[QuotaIds],
+                           properties: Option[Properties]) extends WithAccountId
+
+object JmapQuota {
+
+  val allProperties: Properties = Properties("id", "resourceType", "used", "limit", "scope", "name", "dataTypes", "warnLimit", "softLimit", "description")
+  val idProperty: Properties = Properties("id")
+
+  def propertiesFiltered(requestedProperties: Properties) : Properties = idProperty ++ requestedProperties
+}
+
+
+case class JmapQuota(id: Id,
+                     resourceType: ResourceType,
+                     used: UnsignedInt,
+                     limit: UnsignedInt,
+                     scope: Scope,
+                     name: QuotaName,
+                     dataTypes: List[DataType],
+                     warnLimit: Option[UnsignedInt] = None,
+                     softLimit: Option[UnsignedInt] = None,
+                     description: Option[QuotaDescription] = None)
+
+object QuotaName {
+  def from(scope: Scope, resourceType: ResourceType, dataTypes: List[DataType]): QuotaName =
+    QuotaName(s"${scope.asString()}:${resourceType.asString()}:${dataTypes.map(_.asString()).mkString("_")}")
+}
+
+case class QuotaName(string: String)
+
+case class QuotaDescription(string: String)
+
+sealed trait ResourceType {
+  def asString(): String
+}
+
+case object CountResourceType extends ResourceType {
+  override def asString(): String = "count"
+}
+
+case object OctetsResourceType extends ResourceType {
+  override def asString(): String = "octets"
+}
+
+object Scope {
+  def fromJava(scope: org.apache.james.mailbox.model.Quota.Scope): Scope =
+    scope match {
+      case org.apache.james.mailbox.model.Quota.Scope.Domain => DomainScope
+      case org.apache.james.mailbox.model.Quota.Scope.Global => GlobalScope
+      case org.apache.james.mailbox.model.Quota.Scope.User => AccountScope
+    }
+}
+
+trait Scope {
+  def asString(): String
+
+}
+
+case object AccountScope extends Scope {
+  override def asString(): String = "account"
+}
+
+case object DomainScope extends Scope {
+  override def asString(): String = "domain"
+}
+
+case object GlobalScope extends Scope {
+  override def asString(): String = "global"
+}
```

Comment:
Just for it to be clear:

If I set:

 - Global quota to 1GB
 - Domain quota to 2GB
 - User quota with 5GB
 
 Then `Quota/get` should return a **single** quota with 5GB as a limit!
 
 Is this the case? Can you add a test? (and refactor the code to simplify it?)

Reply:
> Then Quota/get should return a single quota with 5GB as a limit!

Why do we return single 5GB? return Only User quota or return max limit?

Reply:
We only return one limit: the one applied to the user ;-)